File: Sort/countsort.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
vector<int> countSort(vector<int> &arr)
{
    int mini = *min_element(arr.begin(), arr.end());
    for(int i=0;i<arr.size();i++){
        arr[i]-=mini;
    }
    int maxi = *max_element(arr.begin(), arr.end());
    vector<int> freq(maxi + 1, 0);
    for (int &ele : arr)
    {
        freq[ele]++;
    }
    for (int i = 1; i < freq.size(); i++)
    {
        freq[i] = freq[i] + freq[i - 1];
    }
    vector<int> ans(arr.size());
    for (int i = arr.size() - 1; i >= 0; i--)
    {
        ans[--freq[arr[i]]] = arr[i]+mini;
    }
    return ans;
}
int main()
{
    vector<int> arr{-10,0,1,10,-4,7};
    vector<int> ans = countSort(arr);
    for (int &ele : ans)
    {
        cout << ele << " ";
    }
}

File: Sort/countsort.py
#countsort.py
def countSort(arr):
    mini=min(arr)
    for i in range(0,len(arr)):
        arr[i]=arr[i]-mini
    maxi=max(arr)
    freq=[0]*(maxi+1)
    for i in range(0,len(arr)):
        freq[arr[i]]+=1
    
    for i in range(1,len(freq)):
        freq[i]=freq[i]+freq[i-1]
        
    ans=[0]*(len(arr))
    
    for i in range(len(arr) - 1, -1, -1):
        freq[arr[i]] -= 1
        ans[freq[arr[i]]] = arr[i] + mini
    
    return ans
arr=[-10,0,1,10,-4,7]

ans=countSort(arr)

for ele in ans:
    print(ele)